app/src/main/java/com/burhanrashid52/photoediting/EditImageActivity.java

```diff
@@ -273,41 +278,73 @@ private Uri buildFileProviderUri(@NonNull Uri uri) {
                 new File(uri.getPath()));
     }
 
-    @SuppressLint("MissingPermission")
+    @SuppressLint({"MissingPermission",  "NewApi"})
     private void saveImage() {
-        if (requestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+        final String fileName = System.currentTimeMillis() + ".png";
+        if (isSdk29OrHigher()) {
```

Comment:
We can avoid the version check in the activity and move it inside `createFileForSdk29orHigher` method itself and better rename it `createFile`. This will also help us to move the `else` logic to `createFileForSdk29orHigher` making the activity cleaner and aware of how the file is being created.

Reply:
yes this is something must be done